pkg/ipcache/ipcache.go

```diff
@@ -759,11 +761,11 @@ func (ipc *IPCache) GetNamedPorts() (npm types.NamedPortMultiMap) {
 
 // DeleteOnMetadataMatch removes the provided IP to security identity mapping from the IPCache
 // if the metadata cache holds the same "owner" metadata as the triggering pod event.
-func (ipc *IPCache) DeleteOnMetadataMatch(IP string, source source.Source, namespace, name string) (namedPortsChanged bool) {
+func (ipc *IPCache) DeleteOnMetadataMatch(IP string, source source.Source, namespace, name, uid string) (namedPortsChanged bool) {
```

Comment:
Commenting on random like, this refers to line 304:
We probably want to add UID to `scopedLog`:
```
			scopedLog = scopedLog.WithFields(logrus.Fields{
				logfields.K8sPodName:   k8sMeta.PodName,
				logfields.K8sNamespace: k8sMeta.Namespace,
				logfields.NamedPorts:   k8sMeta.NamedPorts,
			})
```